services/dataplatform/nodepool.go

```diff
@@ -0,0 +1,317 @@
+		maintenanceWindowEntry := SetMaintenanceWindowProperties(*nodePool.Properties.MaintenanceWindow)
+		maintenanceWindow = append(maintenanceWindow, maintenanceWindowEntry)
+		if err := d.Set("maintenance_window", maintenanceWindow); err != nil {
+			return utils.GenerateSetError(nodePoolResourceName, "maintenance_window", err)
+		}
+	}
+
+	if nodePool.Properties.Labels != nil {
+		if err := d.Set("labels", *nodePool.Properties.Labels); err != nil {
+			return utils.GenerateSetError(nodePoolResourceName, "storage_size", err)
+		}
+	}
+
+	if nodePool.Properties.Annotations != nil {
+		if err := d.Set("annotations", *nodePool.Properties.Annotations); err != nil {
+			return utils.GenerateSetError(nodePoolResourceName, "annotations", err)
+		}
+	}
+	return nil
+}
+
+func SetNodePoolsData(d *schema.ResourceData, results []dataplatform.NodePoolResponseData) diag.Diagnostics {
+
+	resourceId := uuid.New()
+	d.SetId(resourceId.String())
+
+	if results != nil {
+		var nodePools []interface{}
+		for _, nodePool := range results {
+			nodePoolEntry := make(map[string]interface{})
+			if nodePool.Properties != nil {
+				utils.SetPropWithNilCheck(nodePoolEntry, "name", nodePool.Properties.Name)
+				utils.SetPropWithNilCheck(nodePoolEntry, "data_platform_version", nodePool.Properties.DataPlatformVersion)
+				utils.SetPropWithNilCheck(nodePoolEntry, "datacenter_id", nodePool.Properties.DatacenterId)
+				utils.SetPropWithNilCheck(nodePoolEntry, "node_count", nodePool.Properties.NodeCount)
+				utils.SetPropWithNilCheck(nodePoolEntry, "cpu_family", nodePool.Properties.CpuFamily)
+				utils.SetPropWithNilCheck(nodePoolEntry, "cores_count", nodePool.Properties.CoresCount)
+				utils.SetPropWithNilCheck(nodePoolEntry, "ram_size", nodePool.Properties.RamSize)
+				utils.SetPropWithNilCheck(nodePoolEntry, "availability_zone", nodePool.Properties.AvailabilityZone)
+				utils.SetPropWithNilCheck(nodePoolEntry, "storage_type", nodePool.Properties.StorageType)
+				utils.SetPropWithNilCheck(nodePoolEntry, "storage_size", nodePool.Properties.StorageSize)
+				if nodePool.Properties.MaintenanceWindow != nil {
+					var maintenanceWindow []interface{}
+					maintenanceWindowEntry := SetMaintenanceWindowProperties(*nodePool.Properties.MaintenanceWindow)
+					maintenanceWindow = append(maintenanceWindow, maintenanceWindowEntry)
+					utils.SetPropWithNilCheck(nodePoolEntry, "maintenance_window", maintenanceWindow)
+				}
+				utils.SetPropWithNilCheck(nodePoolEntry, "labels", nodePool.Properties.Labels)
+				utils.SetPropWithNilCheck(nodePoolEntry, "annotations", nodePool.Properties.Annotations)
+			}
+			nodePools = append(nodePools, nodePoolEntry)
+		}
+
+		if nodePools == nil || len(nodePools) == 0 {
+			return diag.FromErr(fmt.Errorf("no nodepools found for criteria, please check your filter configuration"))
+		}
+
+		err := d.Set("node_pools", nodePools)
+		if err != nil {
+			diags := diag.FromErr(fmt.Errorf("while setting node_pools: %w", err))
```

Comment:
@cristiGuranIonos `error while ...` or `error occurred while ...`.

Reply:
no need, wrapping the error will make it clear what happens. adding error is not useful here.